ale_linters/erlang/erlc.vim

```diff
@@ -3,7 +3,8 @@
 let g:ale_erlang_erlc_options = get(g:, 'ale_erlang_erlc_options', '')
 
 function! ale_linters#erlang#erlc#GetCommand(buffer) abort
-    return 'erlc ' . g:ale_erlang_erlc_options . ' %t'
+    let l:temp_dir = has('win32') ? $TMP : $TMPDIR
```

Comment:
This is also how `elm/make.vim` determines the temporary directory. It might be useful to move this logic into a common `ale#utils` function.